debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnectorTask.java

```diff
@@ -90,12 +90,8 @@ public String version() {
 
         schema = new SqlServerDatabaseSchema(connectorConfig, valueConverters, topicSelector, schemaNameAdjuster);
         schema.initializeStorage();
-
-        // TODO: recover from all partitions at once
-        for (SqlServerOffsetContext offsetContext : taskOffsetContext.getOffsets().values()) {
-            if (offsetContext != null) {
-                schema.recover(offsetContext);
-            }
+        if (!taskOffsetContext.isNull()) {
```

Comment:
It's the consumer of `taskOffsetContext` who should perform this check. Otherwise, passing it a non-checked value is unsafe.